### ISSUE TITLE
onlyDifferent input parameter?

### ISSUE THREAD

**Issue by jfaMan** [pressidium/lftp-mirror-action#2]
I have a suggestion. Is it possible to add an input parameter similar to onlyNewer but instead of uploading files with a more recent modification date upload files if the modification date has simply changed?

In my case, I'm using this in a workflow to deploy theme files to WordPress. By default or setting onlyNewer to true speeds up the workflow run time, but the issue I have is if someone was to go into the WordPress theme file editor and directly make changes to a file, that file would now have a more recent modification date to the file in the source repository. This means when running the workflow and deploying, that file won't upload, the changes to that file will persist and will never be in sync until a new commit modifies that source file and makes it 'newer'.

To ensure the theme files on WordPress are always 100% in sync with the source repository, I could set onlyNewer to false but that would increase the workflow run time as now every file has to be uploaded again regardless of whether it has changed or not.

I guess just the possibility of deployed files not actually being deployed in this scenario made me think of this. Maybe it's a bit of an edge case though, so just using onlyNewer false even if it takes longer would be best to avoid the possibility? Just a suggestion anyway.

**Comment by over-engineer** [comment]
Hey, @jfaMan!

That’s a great idea — thanks for taking the time to suggest that.

This was addressed in `v1.1.0` (the `v1` tag also points to the latest commit), by introducing a `restoreMTime` input, which allows you to restore the modification time of the files via git, even when `onlyNewer` is set to `false`.

So, to upload files when the modification date is different, you’ll need to set `onlyNewer` to `false`, and `restoreMTime` to `true`.

For example,

```yaml
- name: Deploy files via SFTP
  uses: pressidium/lftp-mirror-action@v1
  with:
    host: ${{ secrets.SFTP_HOST }}
    port: ${{ secrets.SFTP_PORT }}
    user: ${{ secrets.SFTP_USER }}
    pass: ${{ secrets.SFTP_PASS }}
    onlyNewer: false
    restoreMTime: true
    remoteDir: '/var/www/html/example.com/public'
```

Let me know if you need any clarification 🙂

**Comment by over-engineer** [comment]
Closing as resolved. Feel free to reopen this, if there’s anything I overlooked.
